internal/transport/keepalive_test.go

```diff
@@ -655,44 +669,47 @@ func (s) TestTCPUserTimeout(t *testing.T) {
 	}
 }
 
-func pollForHealthyStream(client *http2Client) error {
+// checkForHealthyStream attempts to create a stream and return error if any.
+// The stream created is closed right after to avoid any leakages.
+func checkForHealthyStream(client *http2Client) error {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
-	var err error
-	for ctx.Err() == nil {
-		if _, err = client.NewStream(ctx, &CallHdr{}); err == nil {
-			break
-		}
-		time.Sleep(50 * time.Millisecond)
-	}
+	stream, err := client.NewStream(ctx, &CallHdr{})
+	client.CloseStream(stream, err)
 	return err
 }
 
 func pollForStreamCreationError(client *http2Client) error {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
-	for ctx.Err() == nil {
+	for {
+		if ctx.Err() != nil {
+			return fmt.Errorf("test timed out before stream creation returned an error")
+		}
 		if _, err := client.NewStream(ctx, &CallHdr{}); err != nil {
-			break
+			return nil
```

Comment:
I believe this is not what we want.  If there is a delay creating a stream, it will be a context expiration error that comes back from here.  And that's exactly the kind of thing we wanted to return as an error from this function, right?

Reply:
Thanks for catching that. You are right, i still want to error on `ctx.Err()`. Updating